### PR TITLE
Create Account struct

### PR DESCRIPTION
**Pull request by zecakeh**
Move account-related methods on `Client` to the new `Account` struct and add new methods.

I cloned the whole client instead of the `HttpClient`, like discussed in the matrix room, to be able to use methods like `upload` or `get_media_content`.

### CODE REVIEW

File: crates/matrix-sdk/src/account.rs [hunk 1]
```diff
@@ -0,0 +1,628 @@
+use std::io::Read;
```
Comment (poljar) [inline crates/matrix-sdk/src/account.rs:1]: The file needs a copyright header.

File: crates/matrix-sdk/src/account.rs [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+    /// # use futures::executor::block_on;
+    /// # use matrix_sdk::Client;
+    /// # use url::Url;
+    /// # block_on(async {
+    /// # let homeserver = Url::parse("http://localhost:8080").unwrap();
+    /// # let client = Client::new(homeserver).await.unwrap();
+    /// let path = Path::new("/home/example/selfie.jpg");
+    /// let mut image = File::open(&path).unwrap();
+    ///
+    /// client.account().upload_avatar(&mime::IMAGE_JPEG, &mut image).await.expect("Can't set avatar");
```
Comment (poljar) [inline crates/matrix-sdk/src/account.rs:185]: This line will now be rendered a bit wide in the docs.

File: crates/matrix-sdk/src/account.rs [hunk 3]
```diff
@@ -184,4 +194,435 @@ impl Account {
...
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - The 3PID being removed.
+    ///
+    /// * `medium` - The type of the 3PID.
+    ///
+    /// * `id_server` - The identity server to unbind from. If not provided, the
+    /// homeserver should unbind the 3PID from the identity server it was bound
+    /// to with [`bind_3pid`].
```
Comment (poljar) [inline crates/matrix-sdk/src/account.rs:584]: Dead link.

File: crates/matrix-sdk/src/account.rs [hunk 4]
```diff
@@ -184,4 +194,435 @@ impl Account {
...
+    /// This method should be called after [`request_3pid_email_token`] or
+    /// [`request_3pid_msisdn_token`] to complete the 3PID registration.
+    ///
+    /// # Arguments
+    ///
+    /// * `client_secret` - The same client secret used in
+    /// [`request_3pid_email_token`] or [`request_3pid_msisdn_token`].
+    ///
+    /// * `sid` - The session ID returned in
+    /// [`request_3pid_email_token`] or [`request_3pid_msisdn_token`].
```
Comment (poljar) [inline crates/matrix-sdk/src/account.rs:549]: Dead links.

File: crates/matrix-sdk/src/account.rs [hunk 5]
```diff
@@ -184,4 +194,435 @@ impl Account {
...
+    }
+
+    /// Add a [Third Party Identifier][3pid] on the homeserver for this
+    /// account.
+    ///
+    /// This 3PID may be used by the homeserver to authenticate the user
+    /// during sensitive operations. To register it against an identity server,
+    /// use [`bind_3pid`].
+    ///
+    /// This method should be called after [`request_3pid_email_token`] or
```
Comment (poljar) [inline crates/matrix-sdk/src/account.rs:540]: Dead link.

File: crates/matrix-sdk/src/account.rs [hunk 6]
```diff
@@ -184,4 +194,435 @@ impl Account {
...
+    /// #     api::client::r0::{
+    /// #         account::change_password::{Request as ChangePasswordRequest},
+    /// #         uiaa::{AuthData, Dummy},
+    /// #     },
+    /// #     assign,
+    /// # };
+    /// # use futures::executor::block_on;
+    /// # use url::Url;
+    /// # block_on(async {
+    /// # let homeserver = Url::parse("http://localhost:8080").unwrap();
```
[the remark this answers is not in the document]
Comment (zecakeh) [reply crates/matrix-sdk/src/account.rs:258]: Sorry about all those unwraps. It's just that I copy-paste from existing examples.

**Review by poljar — APPROVED** [review]

**Pull request merged**